src/main/java/org/example/Main.java

```diff
@@ -17,83 +13,117 @@
-
-                    statement.setString(1, brandName);
-                    statement.setString(2, strength);
-                    statement.setString(3, dosage_form);
-                    statement.execute();
-//                    for (JsonElement brandNameElement : brandName) {
-//                       // String brandName = brandNameElement.getAsString();
-//
-//                        Medication medication = new Medication();
-//                      //  medication.setBrand_name(brandName);
-//                        medications.add(medication);
-//
-//                       // statement.setString(1, brandName);
-//                       // statement.execute();
-//                    }
-                }
-                for (String brand : brandNameList) {
-                    System.out.println(brand);
-                }
-                for(Medication medication : medications){
-                    System.out.println(medication.toString());
-                }
-
-           statement.close();
-            jDbCconnection.close();
-        } else {
-            System.out.println("Error retrieving medication data. Response code: " + responseCode);
+  public static void main(String[] args) throws Exception {
+
+    String apiUrl = "https://api.fda.gov/drug/drugsfda.json?&limit=1000";
+
+    URL url = new URL(apiUrl);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+
+    int responseCode = connection.getResponseCode();
+    if (responseCode == HttpURLConnection.HTTP_OK) {
+      BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+      String inputLine;
+      StringBuffer response = new StringBuffer();
+      while ((inputLine = in.readLine()) != null) {
+        response.append(inputLine);
+      }
+      in.close();
+
+      JsonObject jsonObject = new Gson().fromJson(response.toString(), JsonObject.class);
+      JsonArray results = jsonObject.getAsJsonArray("results");
+
+      Class.forName("org.postgresql.Driver");
+
+      Connection jDbCconnection =
+          DriverManager.getConnection("jdbc:postgresql://localhost:5432/dex");
+      PreparedStatement statement =
+          jDbCconnection.prepareStatement(
+              "INSERT INTO medication_with_uuid (brand_name, strength, dosage_form) VALUES (?,?,?)");
+
+      List<String> brandNameList = new ArrayList<>();
+      List<Medication> medications = new ArrayList<>();
+      for (JsonElement resultElement : results) {
+        JsonObject result = resultElement.getAsJsonObject();
+        // products may be null
+        Optional<JsonArray> maybeProducts = Optional.ofNullable(result.getAsJsonArray("products"));
+        if (maybeProducts.isEmpty()) {
+          continue;
+        }
+        // if we're here that means products are good to go
+        JsonArray products = maybeProducts.get();
+
+        Optional<String> maybeBrandName =
```

Comment:
I think you mentioned Optional but I can't find where. What is Optional used for?

Reply:
So an optional is a container that either has a value in it.. or null. It's main purpose really is to signify that what im doing could potentially come back as a null value.  So here in this example products could be a JsonArray.. could be nothing.. so i'm wrapping that possibility in an Optional. So starting from the overall result object.. there MAY be a product array inside of it.. so im unwrapping the optional and taking the value.. or lack thereof.. and acting on it.

Optional<JsonArray> maybeProduct = Optional.ofNullable(result.getAsJsonArray("products") 

the value of the above line could be null (empty).. if its empty.. im gonna move on to the next iteration

if(maybeProduct.isEmpty()) --> continue

if i get past the above line.. i know for sure that i now have a product array that i can work with.. and now i can use it!

JsonArray product = maybeProduct.get()